plugins/tfidf.cc

```diff
@@ -325,13 +390,23 @@ apply(const FunctionApplier & applier,
         break;
     }
 
+    RowValue values;
+    Date ts = inputVal.getEffectiveTimestamp();
+
     // Compute the score for every word in the input
+    //cerr << "corpus size " << corpusSize << " document size " << documentSize << endl;
+
     for (auto& col : inputVal.getRow() ) {
-        Utf8String colName = std::get<0>(col).toUtf8String();
+        Utf8String term = std::get<0>(col).toUtf8String(); // the term is the columnName
+        double frequency = (double) std::get<1>(col).getAtom().toUInt() / documentSize;
 
-        double frequency = std::get<1>(col).toDouble() / documentCount;
```

Comment:
@mailletf @steadtler Here we divide the term frequency (# of times a term appears in the document) by the document size (the number of terms it contains).  This is a good approach in some cases especially when documents are varying in size greatly but it may not be what the user wants (BM25 is likely a better option).  Since the term frequencies are passed as an input to the tf-idf function, I suggest we do not perform that normalization here and let the user decide if he/she wants to use raw term frequencies or normalized ones.  Do you agree?